### PR TITLE
DEV: Refactor to use Glimmer Components

### PR DESCRIPTION
**Pull request by jancernik**
Since now is possible to render Glimmer components inside posts, it was a good opportunity to use that new functionality in this plugin.

### CODE REVIEW

File: assets/javascripts/discourse/components/lazy-video.hbs [hunk 1]
```diff
@@ -0,0 +1,41 @@
+<div
+  class="{{concat @videoAttributes.providerName '-onebox lazy-video-container'}}
```
Comment (jjaffeux) [inline assets/javascripts/discourse/components/lazy-video.hbs]: you have the `concat-class` helper which would make this a little bit clearer and easier to write

File: assets/javascripts/discourse/components/lazy-video.hbs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+>
+  {{#if this.isLoaded}}
+    <LazyIframe
+      @providerName={{@videoAttributes.providerName}}
+      @title={{@videoAttributes.title}}
+      @videoId={{@videoAttributes.id}}
+    />
+  {{else}}
+    <div
+      class={{concat "video-thumbnail " @videoAttributes.providerName}}
```
Comment (jjaffeux) [inline assets/javascripts/discourse/components/lazy-video.hbs]: concat-class

File: assets/javascripts/discourse/components/lazy-video.hbs [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    </div>
+    <div class="title-container">
+      <div class="title-wrapper">
+        <a
+          target="_blank"
+          rel="noopener noreferrer"
+          class="title-link"
+          href={{@videoAttributes.url}}
+          title={{@videoAttributes.title}}
+        >{{@videoAttributes.title}}</a>
```
Comment (jjaffeux) [inline assets/javascripts/discourse/components/lazy-video.hbs]:
```suggestion
        <a
          target="_blank"
          rel="noopener noreferrer"
          class="title-link"
          href={{@videoAttributes.url}}
          title={{@videoAttributes.title}}
        >
          {{@videoAttributes.title}}
        </a>
```

File: assets/javascripts/initializers/lazy-videos.js [hunk 4]
```diff
@@ -1,23 +1,46 @@
...
       if (cooked.classList.contains("d-editor-preview")) {
         return;
       }
-      lazyVideos.decorateLazyContainers(cooked, api);
+
+      if (lazyContainers.length === 0) {
+        return;
+      }
+
+      for (const container of lazyContainers) {
```
Comment (jjaffeux) [inline assets/javascripts/initializers/lazy-videos.js]: you can just do a simple `forEach` here

File: assets/javascripts/initializers/lazy-videos.js [hunk 5]
```diff
@@ -1,23 +1,46 @@
...
+      const lazyContainers = cooked.querySelectorAll(".lazy-video-container");
+
       if (cooked.classList.contains("d-editor-preview")) {
         return;
       }
-      lazyVideos.decorateLazyContainers(cooked, api);
+
+      if (lazyContainers.length === 0) {
+        return;
+      }
```
Comment (jjaffeux) [inline assets/javascripts/initializers/lazy-videos.js]: with the forEach this won't be needed

File: assets/javascripts/initializers/lazy-videos.js [hunk 6]
```diff
@@ -1,23 +1,46 @@
...
-  const lazyVideos = api.container.lookup("service:lazy-videos");
-
   api.decorateCookedElement(
-    (cooked) => {
+    (cooked, helper) => {
+      const lazyContainers = cooked.querySelectorAll(".lazy-video-container");
+
       if (cooked.classList.contains("d-editor-preview")) {
         return;
       }
```
Comment (jjaffeux) [inline assets/javascripts/initializers/lazy-videos.js:10]: would probably do this before the lazyContainers line, as we don't want to do any work in this case apparently

File: assets/javascripts/lib/data-extractor.js [hunk 7]
```diff
@@ -0,0 +1,15 @@
+import { sanitize } from "discourse/lib/text";
```
Comment (jjaffeux) [inline assets/javascripts/lib/data-extractor.js]: the name of the file might need to be changed. It's also super tailored for `lazy-video-container` when the name might make you think it's a general purpose helper.

File: assets/javascripts/discourse/components/lazy-iframe.hbs [hunk 8]
```diff
@@ -0,0 +1,9 @@
+<iframe
```
Comment (jjaffeux) [inline assets/javascripts/discourse/components/lazy-iframe.hbs]: maybe wrap this in:

```
{{#if @providerName}}
{{/if}}
```

File: assets/javascripts/discourse/components/lazy-video.js [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+
+  @action
+  loadEmbed() {
+    this.isLoaded = true;
+    this.args.callback();
+  }
+
+  @action
+  onKeyPress(event) {
+    if (event.key === "Enter") {
```
Comment (jjaffeux) [inline assets/javascripts/discourse/components/lazy-video.js:18]: maybe we also check `!this.isLoaded` here?

File: assets/javascripts/discourse/components/lazy-video.js [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+import { action } from "@ember/object";
+import { tracked } from "@glimmer/tracking";
+
+export default class LazyVideo extends Component {
+  @tracked isLoaded = false;
+
+  @action
+  loadEmbed() {
+    this.isLoaded = true;
+    this.args.callback();
```
Comment (jjaffeux) [inline assets/javascripts/discourse/components/lazy-video.js]:
```suggestion
    this.args.callback?.();
```

File: assets/javascripts/initializers/lazy-videos.js [hunk 11]
```diff
@@ -1,23 +1,46 @@
...
-      lazyVideos.decorateLazyContainers(cooked, api);
+
+      if (lazyContainers.length === 0) {
+        return;
+      }
+
+      for (const container of lazyContainers) {
+        const callback = () => {
+          const postId = cooked.closest("article").dataset.postId;
```
Comment (jjaffeux) [inline assets/javascripts/initializers/lazy-videos.js]:
```suggestion
          const postId = cooked.closest("article")?.dataset?.postId;
```

File: assets/javascripts/initializers/lazy-videos.js (hunk 6 above)
Comment (jancernik) [reply assets/javascripts/initializers/lazy-videos.js:10]: yeah, I switched them up

**Review by jjaffeux — APPROVED** [review]
👌

**Pull request merged**